python/orca/tutorial/tensorflow/ncf_orca.py

```diff
@@ -0,0 +1,169 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import time
+import numpy as np
+import tensorflow as tf
+import os
+
+from bigdl.orca import OrcaContext, init_orca_context, stop_orca_context
+from bigdl.orca.learn.tf2 import Estimator
+
+
+def model_creator(config):
+    import tensorflow as tf
+    from tensorflow import keras
+    embedding_size=16
+    user = keras.layers.Input(dtype=tf.int32, shape=(None,))
+    item = keras.layers.Input(dtype=tf.int32, shape=(None,))
+    label = keras.layers.Input(dtype=tf.int32, shape=(None,))
+
+    with tf.name_scope("GMF"):
+        user_embed_GMF = keras.layers.Embedding(max_user_id + 1, embedding_size)(user)
+        item_embed_GMF = keras.layers.Embedding(max_item_id + 1, embedding_size)(item)
+        GMF = keras.layers.Multiply()([user_embed_GMF, item_embed_GMF])
+
+    with tf.name_scope("MLP"):
+        user_embed_MLP = keras.layers.Embedding(max_user_id + 1, embedding_size)(user)
+        item_embed_MLP = keras.layers.Embedding(max_item_id + 1, embedding_size)(item)
+        interaction = tf.concat([user_embed_MLP, item_embed_MLP], axis=-1)
+        layer1_MLP = keras.layers.Dense(units=embedding_size * 2, activation='relu')(interaction)
+        layer1_MLP = keras.layers.Dropout(rate=0.2)(layer1_MLP)
+        layer2_MLP = keras.layers.Dense(units=embedding_size, activation='relu')(layer1_MLP)
+        layer2_MLP = keras.layers.Dropout(rate=0.2)(layer2_MLP)
+        layer3_MLP = keras.layers.Dense(units=embedding_size // 2, activation='relu')(layer2_MLP)
+        layer3_MLP = keras.layers.Dropout(rate=0.2)(layer3_MLP)
+
+    # Concate the two parts together
+    with tf.name_scope("concatenation"):
+        concatenation = tf.concat([GMF, layer3_MLP], axis=-1)
+        outputs = keras.layers.Dense(1, activation='sigmoid')(concatenation)
+    
+    model = keras.Model(inputs=[user, item], outputs=outputs)
+    model.compile(optimizer= "adam",
+                  loss= "binary_crossentropy",
+                  metrics=['accuracy'])
+    return model
+
+     
+
+cluster_mode = "local"
+# cluster_mode = "k8s"
+if cluster_mode == "local":
+    sc = init_orca_context(memory="20g")
+elif cluster_mode == "standalone":
+    sc = init_orca_context("standalone", master="spark://...", cores=8, num_nodes=4, memory="10g")
+elif cluster_mode == "yarn":
+    sc = init_orca_context("yarn", cores=8, num_nodes=4, memory="10g")
+elif cluster_mode == "k8s":
+    sc = init_orca_context(cluster_mode="k8s", cores=8, num_nodes=4, memory="20g",
+                           master="k8s://https://172.16.0.200:6443",
```

Comment:
As a tutorial, I think it is better to put the hard-coded parameters in command line options and arguments.

Reply:
right, use argparser

Reply:
do not hardcode k8s master